huggingface/pytorch/tgi/docker/0.5/py3/cu118/Dockerfile.gpu

```diff
@@ -0,0 +1,104 @@
+FROM lukemathwalker/cargo-chef:latest-rust-1.67 AS chef
+WORKDIR /usr/src
+
+FROM chef as planner
+COPY Cargo.toml Cargo.toml
+COPY rust-toolchain.toml rust-toolchain.toml
+COPY proto proto
+COPY router router
+COPY launcher launcher
+RUN cargo chef prepare --recipe-path recipe.json
+
+FROM chef AS builder
+
+RUN PROTOC_ZIP=protoc-21.12-linux-x86_64.zip && \
+    curl -OL https://github.com/protocolbuffers/protobuf/releases/download/v21.12/$PROTOC_ZIP && \
+    unzip -o $PROTOC_ZIP -d /usr/local bin/protoc && \
+    unzip -o $PROTOC_ZIP -d /usr/local 'include/*' && \
+    rm -f $PROTOC_ZIP
+
+COPY --from=planner /usr/src/recipe.json recipe.json
+RUN cargo chef cook --release --recipe-path recipe.json
+
+COPY Cargo.toml Cargo.toml
+COPY rust-toolchain.toml rust-toolchain.toml
+COPY proto proto
+COPY router router
+COPY launcher launcher
+RUN cargo build --release
+
+FROM nvidia/cuda:11.8.0-devel-ubuntu22.04 as base
```

Comment:
We are not using an nvidia image directly see: https://github.com/huggingface/text-generation-inference/blob/e86cca9723d8f7b9a17f11d716240d0ad3e0f875/Dockerfile#L35

Reply:
this is needed for flash attention cuda kernels: https://github.com/huggingface/text-generation-inference/blob/e86cca9723d8f7b9a17f11d716240d0ad3e0f875/Dockerfile#L77

https://github.com/huggingface/text-generation-inference/blob/e86cca9723d8f7b9a17f11d716240d0ad3e0f875/Dockerfile#L88

Reply:
@philschmid do you have and source code lead to the base image you are using? We need to verify the license and agreement holds by these container owners

Reply:
The reason we are using NVIDIA-CUDA image is most of them is cleared from OSS and legal

Reply:
> We are not using an nvidia image directly see: https://github.com/huggingface/text-generation-inference/blob/e86cca9723d8f7b9a17f11d716240d0ad3e0f875/Dockerfile#L35

I was referring to 0.5.0 Dockerfile, which is using nvidia/cuda:11.8.0-devel-ubuntu22.04 image as base: https://github.com/huggingface/text-generation-inference/blob/v0.5.0/Dockerfile#L30

Reply:
The `pytorch-install` base image is just plain `debian:bullseye-slim`

Reply:
The only base images we use are `lukemathwalker/cargo-chef:latest-rust-1.69` and `debian:bullseye-slim`